app/src/main/java/org/sil/storyproducer/model/StoryIO.kt

```diff
@@ -9,36 +9,80 @@ import org.sil.storyproducer.R
 import org.sil.storyproducer.tools.file.*
 import java.io.ByteArrayOutputStream
 import java.io.File
+import timber.log.Timber
+
+
 
 fun Story.toJson(context: Context){
+    val filePath = "$PROJECT_DIR/$PROJECT_FILE" // location of file
     val moshi = Moshi
             .Builder()
             .add(RectAdapter())
             .add(UriAdapter())
             .build()
     val adapter = Story.jsonAdapter(moshi)
     val oStream = getStoryChildOutputStream(context,
-            "$PROJECT_DIR/$PROJECT_FILE","",this.title)
+            filePath,"",this.title)
     if(oStream != null) {
         try {
             oStream.write(adapter.toJson(this).toByteArray(Charsets.UTF_8))
-        }catch(e:java.lang.Exception){}
+        }catch(e:java.lang.Exception){
+            // If we get here, there was an exception thrown while writing the story.json file
+            // Create a suitable error string.  Use  method name, File location, story title & the error
+            val errInfo =   "Method: " + Throwable().stackTrace[0].methodName + ", " +
+                    "File: "   + this.title + ", " +
+                    "Loc: "    + filePath + ", " +
+                    "Err: "    + e.toString()
+
+            // Record the error message & exception (includes stack trace) in FireBase
+            FirebaseCrashlytics.getInstance().log(errInfo)
+            FirebaseCrashlytics.getInstance().recordException(e)
+
+            // Record the message to the android system log
+            // The 4 character tag, "SP::", is a quick way to filter messages in the Logcat utility
+            // found in Android studio
+            // (i.e., to view message during debug, create a Logcat filter for "Log Message:"
+            // looking for "SP::")
+            Timber.e("SP::(%s)", errInfo) // uses Kotlin Log class with severity level: Error
+        }
         finally{oStream.close()}
     }
 }
 
 fun storyFromJson(context: Context, storyTitle: String): Story?{
+    val filePath = "$PROJECT_DIR/$PROJECT_FILE"  // location of file
     try {
+        // use Moshi to restore all information associated with this story
         val moshi = Moshi
                 .Builder()
                 .add(RectAdapter())
                 .add(UriAdapter())
                 .build()
         val adapter = Story.jsonAdapter(moshi)
-        val fileContents = getStoryText(context, "$PROJECT_DIR/$PROJECT_FILE", storyTitle)
+        val fileContents = getStoryText(context, filePath, storyTitle)
                 ?: return null
```

Comment:
Here, a read failure, missing file, or an empty file causes an immediate return null, without reporting any problem. Seems okay? I can't think of how such a report would be useful to us. It would likely occur after a Story.toJson exception was already reported. Also Robin has in the past instructed a user to delete story.json after that project became corrupted. Any further thoughts?

Reply:
Ok, I'm wrong. Since this is within a try{} an exception within getStoryText() will be caught and reported, right? Which of the circumstances will actually throw an exception there? Read failure and missing file?
An empty file might succeed here, return an empty string, but adapter.fromJson should then have an exception, properly. Any further thoughts?

Reply:
I know adapter.fromJson throws.  For a non existent file, it looks like getStoryText returns a null which will percolate through  storyFromJason causing a new story.json file to be created.  If a story.json file exists and there is no data in it, I would say that should not happen and report it as an exception.  Something truncated the file maybe causing loss of data.